api/type.rb

```diff
@@ -474,6 +455,61 @@ def property_ns_prefix
         'Property'
       ]
     end
+
+    # Responsible for handling the logic involved with a single ResourceRef.
+    # A ResourceRef is made up of one-or-more of these Individual ResourceRefs.
+    class IndividualResourceRef < Type
```

Comment:
Since `ResourceRef` is just a collection now, would it make sense to rename that to `ResourceRefs` and then call this `ResourceRef` instead of `IndividualResourceRef`?

Reply:
I wanted to avoid renaming ResourceRef because we reference that class name in a lot of places and this PR was already getting big.

Not a great reason. Happy to reconsider.

Reply:
I'd also appreciate it if you'd do it, big PR notwithstanding.